src/Actions/BcgovSettings.php

```diff
@@ -0,0 +1,166 @@
+
+    /**
+     * Registers BCGov Block Theme settings and sanitizes user input.
+     *
+     * @since 1.1.2
+     * @return void
+     */
+    public function bcgov_register_settings() {
+        // Register callbacks to sanitize user input:
+        // For BCGov Admin Notification settings.
+        register_setting(
+            'bcgov-settings-group',
+            'bcgov_admin_notification_settings',
+            [ $this, 'sanitize_bcgov_admin_notification_settings' ]
+        );
+
+        // For BCGov Custom Pattern settings.
+        register_setting(
+            'bcgov-settings-group',
+            'bcgov_custom_patterns_settings',
+            [ $this, 'sanitize_bcgov_custom_patterns_settings' ]
+        );
+
+        // Add settings sections:
+        // For BCGov Admin Notification settings.
+        add_settings_section(
+            'bcgov_admin_notification_settings',
+            'Appearance > Custom Notice Banner',
+            [ $this, 'bcgov_admin_notification_settings' ],
+            'bcgov-theme-settings'
+        );
+
+        // For BCGov Custom Pattern settings.
+        add_settings_section(
+            'bcgov_custom_patterns_settings',
+            'Custom Post Type > Patterns',
+            [ $this, 'bcgov_custom_patterns_settings' ],
+            'bcgov-theme-settings'
+        );
+    }
+
+    /**
+     * Sanitize and save the Custom Pattern checkbox value.
+     *
+     * @param array $input The settings input.
+     * @return array The sanitized input.
+     */
+    public function sanitize_bcgov_custom_patterns_settings( $input ) {
+        // Sanitize the checkbox value.
+        $input['show_custom_patterns'] = isset( $input['show_custom_patterns'] ) ? 1 : 0;
+        return $input;
+    }
+
+    /**
+     * Sanitize and save the Admin Custom Notification checkbox value.
+     *
+     * @param array $input The settings input.
+     * @return array The sanitized input.
+     */
+    public function sanitize_bcgov_admin_notification_settings( $input ) {
+        // Sanitize the checkbox value.
+        $input['show_admin_notification'] = isset( $input['show_admin_notification'] ) ? 1 : 0;
+        return $input;
+    }
+
+    /**
+     * Displays the Admin Custom Notification settings section description.
+     * Includes a checkbox for enabling/disabling the display of Admin Custom Notifications.
+     *
+     * @since 1.1.2
+     * @return void
+     */
+    public function bcgov_admin_notification_settings() {
+         $value                  = get_option( 'bcgov_admin_notification_settings' ) ?? [];
+        $show_admin_notification = isset( $value['show_admin_notification'] ) ? $value['show_admin_notification'] : 0;
+		?>
+        <input type="checkbox" name="bcgov_admin_notification_settings[show_admin_notification]" id="bcgov_admin_notification" value="1" <?php checked( $show_admin_notification, 1 ); ?> />
+        <label for="bcgov_admin_notification">Check to show the Custom Notice Banner tool</label>
```

Comment:
should probably be using `__()` function
```suggestion
        <label for="bcgov_admin_notification"><?php __('Check to show the Custom Notice Banner tool');?></label>
```